nephio-webui/gen-configmap.yaml

```diff
@@ -0,0 +1,79 @@
+apiVersion: fn.kpt.dev/v1alpha1
+kind: GenConfigMap
+metadata:
+  name: gen-app-config
+  annotations:
+    config.kubernetes.io/local-config: "true"
+configMapMetadata:
+  name: nephio-webui-config
+  namespace: nephio-webui
+params:
+  scheme: http
+  hostname: localhost
+  port: "7007"
+  resourcesNamespace: default
+  authProvider: oidc
+data:
+- type: gotmpl
+  key: app-config.nephio.yaml
+  value: |
+    app:
+      baseUrl: {{.scheme}}://{{.hostname}}:{{.port}}
+
+    # Backstage backend configuration
+    backend:
+      # Note that the baseUrl should be the URL that the browser and other clients
+      # should use when communicating with the backend, i.e. it needs to be
+      # reachable not just from within the backend host, but from all of your
+      # callers. When its value is "http://localhost:7007", it's strictly private
+      # and can't be reached by others.
+      baseUrl: {{.scheme}}://{{.hostname}}:{{.port}}
+
+      listen:
+        port: {{.port}}
+
+      # Content Security Policy
+      csp:
+        # Allows images to be pulled from GitHub and Nepio
+        img-src: ["'self'", 'data:']
+        upgrade-insecure-requests: false
+
+    auth:
+      # see https://backstage.io/docs/auth/ to learn about auth providers
+      providers:
+        google:
+          development:
+            clientId: ${AUTH_GOOGLE_CLIENT_ID}
+            clientSecret: ${AUTH_GOOGLE_CLIENT_SECRET}
+        okta:
+          development:
+            clientId: ${AUTH_OKTA_CLIENT_ID}
+            clientSecret: ${AUTH_OKTA_CLIENT_SECRET}
+            audience: ${AUTH_OKTA_AUDIENCE}
+            authServerId: ${AUTH_OKTA_AUTH_SERVER_ID}
+            idp: ${AUTH_OKTA_IDP}
+
+    # Config as Data Plugin configuration
+    configAsData:
+      # The namespace where Porch managed resources live.
+      resourcesNamespace: {{.resourcesNamespace}}
+
+      # Do not use Config Sync
+      gitOpsDeliveryTool: none
+
+      # Management cluster configuration
+      clusterLocatorMethod:
+        authProvider: ${AUTH_PROVIDER}
+
+        # For authProvider: oidc
+        oidcTokenProvider: ${AUTH_OIDC_TOKEN_PROVIDER}
+
+        # For authProvider: service-account
+        serviceAccountToken: ${AUTH_SERVICE_ACCOUNT_TOKEN}
+
+      # Nephio branding customizations
+      branding:
+        title: Nephio
+        header:
+          logoUrl: {{.scheme}}://{{.hostname}}:{{.port}}/nephio_logo_colorwhite_horizontal.svg
```

Comment:
Do not bring the branding locally, rather point to github directly

```
          logoUrl: https://raw.githubusercontent.com/nephio-project/kpt-backstage-plugins/v1.0.1/packages/app/public/nephio_logo_colorwhite_horizontal.svg
          backgroundImageUrl: https://raw.githubusercontent.com/nephio-project/kpt-backstage-plugins/v1.0.1/packages/app/public/nephio-background.png
```

Reply:
Hmm. I had it that way in the workshop environment. But it would break in an on-prem without Internet. Of course so would a million other things...

Why do you want to change that?

Reply:
When using an Ingress / Gateway to expose the nephio-webui service, we end up with CORS issue when the app tries to resolves the image. I tried enabling CORS in the UI, but image resolution still failed with CORS error. Pointing to remote, internet accessible, images solves this (with the addition of the CSP config added above).

Also, I don't think there is any expectation to run Nephio disconnected for now.